### PR TITLE
Fix: terminal not properly restored on exit + strerr bombing prevention.

### PR DESCRIPTION
**Pull request by gbin**
The terminal issues became a huge pain with the simulators. This should be fixing the corruptions from stderr and retoring the terminal.

### CODE REVIEW

**Comment by mik90** [comment]
Just pulled this down and tested it out on my laptop running fedora + bash in the vscode integrated terminal. Exit works well now!

I did test out `killall cu-caterpillar` (should default to SIGTERM)  out of sheer curiosity and rudeness, and it didn't handle that well. It just prints a bunch of ANSI color codes. Not important tho, this is already an improvement

File: components/monitors/cu_consolemon/src/lib.rs [hunk 1]
```diff
@@ -585,9 +577,29 @@ impl CuMonitor for CuConsoleMon {
...
+            let backend = CrosstermBackend::new(stdout());
+
+            setup_terminal();
+
+            // nukes stderr into orbit as anything in the stack can corrupt the terminal
+            let dev_null = File::open("/dev/null").expect("Could not get /dev/null");
+            let dev_null_fd = dev_null.as_raw_fd();
+            let original_stderr_fd = unsafe { dup(2) };
+            unsafe {
+                dup2(dev_null_fd, 2);
```
Comment (mik90) [inline components/monitors/cu_consolemon/src/lib.rs:589]: Does the libc create have a constant for stderr or do you have to use `2`? If not, could be good to use a var to denote that. I had to double check https://en.wikipedia.org/wiki/File_descriptor

File: components/monitors/cu_consolemon/src/lib.rs [hunk 2]
```diff
@@ -664,7 +669,17 @@ fn init_error_hooks() {
     }));
 }
 
+fn setup_terminal() {
+    enable_raw_mode()
+        .map_err(|e| CuError::new_with_cause("Could not enable console raw mode", e))
+        .expect("Could not enter raw mode: check terminal compatibility.");
```
Comment (mik90) [inline components/monitors/cu_consolemon/src/lib.rs:675]: Do we need both map_err and expect here? I think the latter would do

**Review by mik90 — APPROVED** [review]

File: components/monitors/cu_consolemon/src/lib.rs (hunk 2 above)
Comment (gbin) [reply components/monitors/cu_consolemon/src/lib.rs:675]: oh yeah, indeed a brainless copy paste / autocomplete line lol

File: components/monitors/cu_consolemon/src/lib.rs (hunk 1 above)
Comment (gbin) [reply components/monitors/cu_consolemon/src/lib.rs:589]: ha yes, libc::STDERR_FILENO  it is

**Pull request merged**